python_tests/test_csv_download.py

```diff
@@ -0,0 +1,123 @@
+from typing import Any
+
+import optuna
+from optuna.trial import TrialState
+from optuna_dashboard._app import create_app
+import pytest
+
+from .wsgi_client import send_request
+
+
+def test_download_csv_no_trial() -> None:
+    def objective(trial: optuna.Trial) -> float:
+        x = trial.suggest_float("x", -100, 100)
+        y = trial.suggest_categorical("y", [-1, 0, 1])
+        return x**2 + y
+
+    storage = optuna.storages.InMemoryStorage()
+    study = optuna.create_study(storage=storage)
+    study.optimize(objective, n_trials=0)
+    app = create_app(storage)
+    status, _, body = send_request(
+        app,
+        "/csv/0",
+        "GET",
+        content_type="application/json",
+    )
+    assert status == 200
+
+
+def test_download_csv_all_waiting() -> None:
+    storage = optuna.storages.InMemoryStorage()
+    study = optuna.create_study(storage=storage)
+    study.add_trial(optuna.trial.create_trial(state=TrialState.WAITING))
+    app = create_app(storage)
+    status, _, body = send_request(
+        app,
+        "/csv/0",
+        "GET",
+        content_type="application/json",
+    )
+    assert status == 200
+
+
+def test_download_csv_all_running() -> None:
+    storage = optuna.storages.InMemoryStorage()
+    study = optuna.create_study(storage=storage)
+    study.add_trial(optuna.trial.create_trial(state=TrialState.RUNNING))
+    app = create_app(storage)
+    status, _, body = send_request(
+        app,
+        "/csv/0",
+        "GET",
+        content_type="application/json",
+    )
+    assert status == 200
+
+
+@pytest.mark.parametrize("id", [0, 1])
+def test_download_csv_fail(id: int) -> None:

```

Comment:
`id` is a reserved word in Python, so better avoiding it.
```suggestion
@pytest.mark.parametrize("study_id", [0, 1])
def test_download_csv_fail(study_id: int) -> None:
```